infomapcog/distances.py

```diff
@@ -73,32 +73,57 @@ def needleman_wunsch(x, y, lodict={}, gop=-2.5, gep=-1.75):
     'lodict' must be a dictionary with all symbol pairs as keys and
     match scores as values, or a False value (including an empty
     dictionary) to denote (-1, 1) scores. gop and gep are gap
-    penalties for opening/extending a gap.
+    penalties for opening/extending a gap; alternatively, you can set
+    'gop' to None and provide element/gap alignment costs.
 
     Returns the alignment score and one optimal alignment.
+    
+    >>> needleman_wunsch("AAAAABBBB", "AACAABBCB")
+    (5.0, [('A', 'A'), ('A', 'A'), ('A', 'C'), ('A', 'A'), ('A', 'A'), ('B', 'B'), ('B', 'B'), ('B', 'C'), ('B', 'B')])
+    >>> needleman_wunsch("banana", "mancala", local=True)
```

Comment:
Well, you may just want to split up all functions for alignments. Even if it looks redundant, in my experience, it is quite hard to combine all aspects properly, as you will add complexity even to those simpler functions. I recommend to have a look at the instructions I give in [sequencecomparison.github.io](sequencecomparison.github.io), which lists the major modes, gap extension, etc. all in pseudo-code (or compare with lingpy's low-level cython code). Because it won't stop with local aligments: semi-global, dialign, etc., there are many more ways...